src/err/mod.rs

```diff
@@ -439,15 +455,16 @@ impl PyErr {
         T: ToPyObject,
     {
         fn inner(err: &PyErr, py: Python<'_>, exc: PyObject) -> bool {
-            (unsafe { ffi::PyErr_GivenExceptionMatches(err.type_ptr(py), exc.as_ptr()) }) != 0
+            (unsafe { ffi::PyErr_GivenExceptionMatches(err.get_type(py).as_ptr(), exc.as_ptr()) })
+                != 0
         }
         inner(self, py, exc.to_object(py))
     }
 
     /// Returns true if the current exception is instance of `T`.
     #[inline]
     pub fn is_instance(&self, py: Python<'_>, ty: &PyAny) -> bool {
-        (unsafe { ffi::PyErr_GivenExceptionMatches(self.type_ptr(py), ty.as_ptr()) }) != 0
+        (unsafe { ffi::PyErr_GivenExceptionMatches(self.get_type(py).as_ptr(), ty.as_ptr()) }) != 0
```

Comment:
Changing these lines is probably a bug fix, because otherwise it uses the unnormalized exception type pointer for instance checks, and `self.value(py)` or `self.get_type(py)` may then return completely unrelated stuff if normalization fails.

I'll split this out into a separate PR for communication clarity.